zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -133,14 +147,27 @@ where
             address_book,
             peer_service,
             next_peer_min_wait: sleep(Duration::from_secs(0)),
+            next_update_time: Instant::now(),
         }
     }
 
     /// Update the peer set from the network, using the default fanout limit.
     ///
     /// See [`update_initial`][Self::update_initial] for details.
+    ///
+    /// ## Security
+    ///
+    /// This call is rate-limited to prevent sending a burst of repeated requests for new peer
+    /// addresses to the same peers. Each call will only update the [`CandidateSet`] if more time
+    /// than [`MIN_PEER_GET_ADDR_INTERVAL`][Self::MIN_PEER_GET_ADDR_INTERVAL] has passed since the
+    /// last call.
     pub async fn update(&mut self) -> Result<(), BoxError> {
-        self.update_timeout(None).await
+        if self.next_update_time <= Instant::now() {
+            self.update_timeout(None).await?;
+            self.next_update_time = Instant::now() + Self::MIN_PEER_GET_ADDR_INTERVAL;
+        }
+
+        Ok(())
```

Comment:
Please move this check into `update_timeout`. Otherwise, Zebra can call `update_initial`, then immediately do a redundant `update`.

Also please add a test that calls `update_initial` then `update`, and check that the `update` is skipped.